lib/discourse_redis.rb

```diff
@@ -150,6 +150,26 @@ def self.new_redis_store
     Cache.new
   end
 
+  def multi
+    if block_given?
+      @redis.multi do |transaction|
+        yield DiscourseRedis.new(@config, namespace: @namespace, raw_redis: transaction)
+      end
+    else
+      @redis.multi
+    end
+  end
+
+  def pipelined
+    if block_given?
+      @redis.pipelined do |transaction|
+        yield DiscourseRedis.new(@config, namespace: @namespace, raw_redis: transaction)
+      end
+    else
+      @redis.pipelined
+    end
+  end
```

Comment:
```suggestion
  def pipelined(&block)
    @redis.pipelined do |transaction|
      yield DiscourseRedis.new(@config, namespace: @namespace, raw_redis: transaction)
    end
  end
```

Reply:
sadly I think that would break the deprecation message cause we are unconditionally simulating a block.

Reply:
Calling `@redis.pipelined` is actually going to throw an error if a block is not passed to the method. My suggested changes maintains the same behaviour without the need for us to manually check if a block is given.

Reply:
hmmm not sure I am following... we are unconditionally calling .pipelined with a block under this suggestion? 

```
irb(main):003:0> def a(&blk); end
=> :a
irb(main):004:0> a
=> nil
```

Reply:
But we're also unconditionally yielding so the error raised will be the same with the suggestion. 

```
  def pipelined2
    if block_given?
      @redis.pipelined do |transaction|
        yield DiscourseRedis.new(@config, namespace: @namespace, raw_redis: transaction)
      end
    else
      @redis.pipelined
    end
  end

  def pipelined(&block)
    @redis.pipelined do |transaction|
      yield DiscourseRedis.new(@config, namespace: @namespace, raw_redis: transaction)
    end
  end
```

```
tgxworld ~/work/discourse* fix-pipelined-multi $ rails c
Loading development environment (Rails 7.0.2.4)
[1] pry(main)> Discourse.redis.pipelined2
LocalJumpError: no block given (yield)
from /home/tgxworld/.asdf/installs/ruby/2.7.5/lib/ruby/gems/2.7.0/gems/redis-4.5.1/lib/redis.rb:2670:in `block in pipelined'
[2] pry(main)> Discourse.redis.pipelined
LocalJumpError: no block given (yield)
from /home/tgxworld/work/discourse/lib/discourse_redis.rb:175:in `block in pipelined'
[3] pry(main)> 
```

Reply:
Btw, this is really minor so we don't really have to make the change as part of the PR. I just thought it'll be nice if we can simplify the method.

Reply:
I am confused cause of this source code... 

https://github.com/redis/redis-rb/blob/674541c5cc162b46066556811967a28a184f4665/lib/redis.rb#L165-L168

I guess this was added AFTER the redis we are using, so I will keep this a bit more verbose code cause later on it will give us nicer deprecation + local jump error.